specs/lib/addDefaultsSpec.js

```diff
@@ -244,6 +245,14 @@ describe('addDefaults', function() {
                 expect(gltfClone.materials[materialID].values).toEqual(expectValues);
             }
         }
+
+        // check that shader source is in extras._pipeline.source
+        var shaders = gltfClone.shaders;
+        expect(Object.keys(shaders).length > 0).toEqual(true);
+        for (var shaderID in shaders) {
+            var shader = shaders[shaderID];
```

Comment:
When looping over keys, it's safer to check for `hasOwnProperty` like https://github.com/AnalyticalGraphicsInc/gltf-pipeline/blob/master/lib/addDefaults.js#L19-L21.